controllers/argocd/dex.go

```diff
@@ -229,10 +229,9 @@ func (r *ReconcileArgoCD) reconcileDexDeployment(cr *argoprojv1a1.ArgoCD) error
 			"/shared/argocd-dex",
 			"rundex",
 		},
-		Image:           getDexContainerImage(cr),
-		ImagePullPolicy: corev1.PullAlways,
```

Comment:
Removing the "PullAlways" as default lgtm but do you want to provide an option to the users if they want to use "PullAlways" ?

Also please fix the failing e2e.

Reply:
Thanks, so the behaviour will be if the image tag is :latest or empty - the imagePullPolicy will set to 'Always' but if the tag is specified then it will be 'IfNotPresent'.

Reply:
Default image pull policy
When you (or a controller) submit a new Pod to the API server, your cluster sets the imagePullPolicy field when specific conditions are met:

if you omit the imagePullPolicy field, and the tag for the container image is :latest, imagePullPolicy is automatically set to Always;
if you omit the imagePullPolicy field, and you don't specify the tag for the container image, imagePullPolicy is automatically set to Always;
if you omit the imagePullPolicy field, and you specify the tag for the container image that isn't :latest, the imagePullPolicy is automatically set to IfNotPresent.